### PR TITLE
User/amohdsharfuddin/dkip 47

### PR DESCRIPTION
**Pull request by amohdsharfuddin**
Added functionality for reading excel file and return as JSON. Data fetched is tored in Flux Store.

- Current push only fetches and stores the data.
- Data can be logged using `console.log(Store.getVocabItems()).`
- Data structure is flattened out.
- New packages implemented:
    - axios
    - xlsx
- Data is fetched from top level (`index.js`) and dispatched to Flux Store.

### CODE REVIEW

File: src/data/categories/readExcel.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+import axios from "axios";
+
+const fetchData = async (url) => {
+  const data = await axios({
+    method: "get",
+    url: url,
+    responseType: "blob",
+  })
+    .then((res) => res.data)
+    .catch((err) => console.log(err));
```
Comment (kst9395) [inline src/data/categories/readExcel.js:11]: console.log is fine for development but might want to consider removing it once the product is ready and replaced it with something more meaningful.

Comment (amohdsharfuddin) [reply src/data/categories/readExcel.js:11]: perhaps redirecting to an error page would be better? currently I am editting the error page to display "Coming Soon". perhaps having both error page and coming soon page would be good. what do you think?

**Review by haczuga — CHANGES_REQUESTED** [review]
image
The data structure is changed causing the menu broken

**Review by amohdsharfuddin — COMMENTED** [review]
The sidebar issue has been resolved in the next pull request.

**Pull request merged**